crates/vim/src/normal.rs

```diff
@@ -141,7 +143,28 @@ pub(crate) fn register(workspace: &mut Workspace, cx: &mut ViewContext<Workspace
         Vim::update(cx, |vim, cx| {
             vim.record_current_action(cx);
             vim.update_active_editor(cx, |_, editor, cx| {
-                editor.transact(cx, |editor, cx| editor.indent(&Default::default(), cx))
+                editor.transact(cx, |editor, cx| {
+                    let mut original_positions: HashMap<_, _> = Default::default();
+                    let (map, selections) = editor.selections.all_display(cx);
+                    selections.iter().for_each(|selection| {
+                        let line_start = first_non_whitespace(&map, true, selection.start).column();
+                        original_positions.insert(selection.id, (selection.start, line_start));
+                    });
+                    editor.indent(&Default::default(), cx);
+                    editor.set_clip_at_line_ends(true, cx);
+                    editor.change_selections(Some(Autoscroll::fit()), cx, |s| {
+                        s.move_with(|map, selection| {
+                            let (mut cursor, original_line_start) =
+                                original_positions.remove(&selection.id).unwrap();
+                            *cursor.column_mut() +=
```

Comment:
Do we have to update the column? Vim doesn't seem to do it:

[Screencast from 2024-06-03 17-04-37.webm](https://github.com/zed-industries/zed/assets/1185253/1b283152-487c-4fb2-9c62-0d17ccc38e63)

Reply:
Is it nvim ? Vim seems to make the cursor stays on first column.
[vim-indent-visual-line.webm](https://github.com/zed-industries/zed/assets/8695142/b0be2a52-b7b8-4029-812d-f55ec4ef8173)

Reply:
Yeah, seems like it's an nvim thing. With `nvim -u DEFAULTS`

https://github.com/zed-industries/zed/assets/1185253/0516b4c5-b8ab-47c4-bd2c-78ff65033385

@ConradIrwin @baldwindavid what do you think?

Reply:
Well, nvim seems to do things in its own way...
[nvim-indent-visual-line.webm](https://github.com/zed-industries/zed/assets/8695142/4548ba1b-fb52-4a77-87d2-75ef113e1d8e)

Reply:
To me, having the cursor stay with the text was pleasant. But I'm quite new to vim motions so I've not yet forged old habits. Long time user advice could help indeed.
Otherwise I'm ok with your changes.

Reply:
I don't have a strong opinion on it so would defer to anyone that does, but think it's easy to justify the simpler approach given that it's a neovim default. My sense is that neovim is more common these days than vim for those using it as their main editor.

Reply:
@p-A-uLo then I'd say let's go with the simpler approach. Do you want to use the code I sketched out? I think we could probably also put it in two functions: `let original_positions = save_selection_starts(editor)` and `restore_selection_cursors(editor, original_positions)`.

Reply:
Sorry for the slow response here. I think it's better to adjust to match nvim (but fine to match vim too), either way we need to use an approach based on anchors not display points (as mentioned in the other PR).

The problem with setting and restoring DisplayPoints is that changing the text may invalidate the point. For example in the string "hållo" - you cannot use DisplayPoint::new(0, 2) because it is in the middle of the å. If you store the anchor and then restore it, you'll get nvim like behavior "for free".